src/overlays/actors/ovl_Obj_Kendo_Kanban/z_obj_kendo_kanban.c

```diff
@@ -217,238 +233,251 @@ void ObjKendoKanban_Destroy(Actor* thisx, PlayState* play) {
-        this->unk_2CC = D_80B66690;
-
-        this->unk_29C[0] = D_80B666D0;
-        this->unk_29C[1] = D_80B666DC;
-        this->unk_29C[2] = D_80B66724;
-        this->unk_29C[3] = D_80B66718;
-    } else if (this->unk_30C == (OBJKENDOKANBAN_F_2 | OBJKENDOKANBAN_F_1)) {
-        this->unk_304 = 0x71C;
-        this->actor.velocity = D_80B66684;
-        this->unk_2CC = D_80B666B4;
-
-        this->unk_29C[0] = D_80B666D0;
-        this->unk_29C[1] = D_80B666E8;
-        this->unk_29C[2] = D_80B6670C;
-        this->unk_29C[3] = D_80B666F4;
+
+    } else if (this->boardFragments == LEFT_HALF) {
+        // Initialize the newly spawned left half
+        this->rotationVelocity = 0x71C; // 10 degrees
+        this->actor.velocity = sVelocityLeftHalf;
+        this->centerPoint = sCenterPointLeftHalf;
+
+        this->cornerPoints[0] = sPointTL;
+        this->cornerPoints[1] = sPointTC;
+        this->cornerPoints[2] = sPointBC;
+        this->cornerPoints[3] = sPointBL;
+
+    } else if (this->boardFragments == TOP_HALF) {
+        // Initialize the newly spawned top half
+        this->rotationVelocity = 0x71C; // 10 degrees
+        this->actor.velocity = sVelocityTopHalf;
+        this->centerPoint = sCenterPointTopHalf;
+
+        this->cornerPoints[0] = sPointTL;
+        this->cornerPoints[1] = sPointTR;
+        this->cornerPoints[2] = sPointCR;
+        this->cornerPoints[3] = sPointCL;
     }
 
     this->unk_30A = 0;
-    this->actionFunc = func_80B65CE0;
+    this->actionFunc = ObjKendoKanban_Aerial;
 }
 
-void func_80B65CE0(ObjKendoKanban* this, PlayState* play) {
+void ObjKendoKanban_Aerial(ObjKendoKanban* this, PlayState* play) {
     this->actor.velocity.y += this->actor.gravity;
     Actor_UpdatePos(&this->actor);
-    this->unk_302 += this->unk_304;
-    func_80B65DA8(this, play);
+    this->rotationAngle += this->rotationVelocity;
+    ObjKendoKanban_HandlePhysics(this, play);
     if (this->actor.world.pos.y < -200.0f) {
         this->actor.world.pos.y = -200.0f;
     }
 }
 
-void func_80B65D54(ObjKendoKanban* this) {
-    this->actionFunc = func_80B65D68;
+void ObjKendoKanban_SetupSettled(ObjKendoKanban* this) {
+    this->actionFunc = ObjKendoKanban_Settled;
 }
 
-void func_80B65D68(ObjKendoKanban* this, PlayState* play) {
+void ObjKendoKanban_Settled(ObjKendoKanban* this, PlayState* play) {
     Player* player = GET_PLAYER(play);
 
-    if (func_80B6618C(this, play) == 1) {
-        player->unk_AC0 = 700.0f;
+    if (ObjKendoKanban_IsPlayerOnTop(this, play) == true) {
+        player->displacementY = 700.0f;
     }
 }
 
-void func_80B65DA8(ObjKendoKanban* this, PlayState* play) {
-    Vec3f sp5C;
+void ObjKendoKanban_HandlePhysics(ObjKendoKanban* this, PlayState* play) {
+    Vec3f rootCornerPos = this->cornerPos[0];
     s32 pad[2];
-    Vec3f sp48;
-    s32 pad2;
-    s32 index = 0;
+    Vec3f vecCenterOut = this->actor.world.pos;
+    const s16 MAX_BOUNCE_COUNT = 7;
+    s32 idxRootCornerPos = 0;
```

Comment:
```suggestion
    s32 indexRootCornerPos = 0;
```
We prefer `index` over `idx` (`idx` can sometimes get confused with `id`)